src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -12,13 +69,494 @@ import { ActivatedRoute } from '@angular/router';
+          }
+
+          if (this.isDisplayArchive === undefined) {
+            this.statusMessageService.showErrorMessage('Error retrieving indicator for showing archived courses');
+          }
+        }, (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorMessage(resp.error.message);
+        });
+  }
+
+  /**
+   * Change value of the checkbox Display Archive Courses.
+   */
+  toggleDisplayArchive(): void {
+    this.isDisplayArchive = !this.isDisplayArchive;
+    const isDisplayArchiveChecked: boolean = this.isDisplayArchive;
+    this.courses.forEach((course: Course) => {
+      if (course && course.isArchived) {
+        this.toggleCourseStateAtInput(course, isDisplayArchiveChecked);
+      }
+    });
+  }
+
+  /**
+   * Function to change states of all courses.
+   */
+  toggleAllPresentCoursesStateAtInput(defaultState: boolean): void {
+    this.allPresentCourses.forEach((course: Course) => {
+      if (course) {
+        this.toggleCourseStateAtInput(course, defaultState);
+      }
+    });
+  }
+
+  /**
+   * Change the state of all sections presented on the screen.
+   */
+  toggleAllPresentSectionDetailsState(defaultState: boolean): void {
+    this.allPresentSectionDetails.forEach((sectionDetails: SectionDetails) => {
+      this.toggleSectionState(sectionDetails, defaultState);
+    });
+  }
+
+  /**
+   * Change the state of all teams presented on the screen.
+   */
+  toggleAllPresentTeamDetailsState(defaultState: boolean): void {
+    this.allPresentTeamDetails.forEach((teamDetails: TeamDetails) => {
+      this.toggleTeamState(teamDetails, defaultState);
+    });
+  }
+
+  /**
+   * Change the state of all students presented on the screen.
+   */
+  toggleAllPresentStudentDetailsState(defaultState: boolean): void {
+    this.allPresentStudentDetails.forEach((studentDetails: StudentDetails) => {
+      this.toggleStudentState(studentDetails, defaultState);
+    });
+  }
+
+  /**
+   * Function to trigger getting CourseDetails data for a course from backend.
+   */
+  toggleCourseStateAtInput(course: Course, designatedState?: boolean): void {
+    if (course.isChecked === undefined) {
+      course.isChecked = false;
+    }
+    const state: boolean = designatedState === undefined
+        ? !course.isChecked
+        : designatedState;
+
+    this.fetchCourseDetails(course.id)
+        .subscribe((courseDetails: CourseDetails | null) => {
+          if (courseDetails) {
+            this.toggleCourseState(courseDetails, state);
+            this.courseStudentListSectionDataMap[courseDetails.id]
+                = this.getStudentListSectionDataFromCourseDetails(courseDetails);
+          }
+        }, (error: any) => {
+          this.statusMessageService.showErrorMessage(error);
+        });
+  }
+
+  /**
+   * Function to call backend API to get CourseDetails data for a course.
+   */
+  fetchCourseDetails(courseid: string): Observable<CourseDetails | null> {
```

Comment:
Do you really need a return value for this function?

Reply:
yes i do. Because this function is to retrieve course data from backend, and hence we need to perform a check whether to carry out such retrieval. Processing of the retrieved data should be outside this function (see [line 276](https://github.com/TEAMMATES/teammates/pull/9374/files#diff-3e7add916ff8b5693f7e8cb03b522b9aR276))